junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledOnOsCondition.java

```diff
@@ -23,19 +22,34 @@
  */
 class EnabledOnOsCondition extends BooleanExecutionCondition<EnabledOnOs> {
 
-	static final String ENABLED_ON_CURRENT_OS = "Enabled on operating system: " + System.getProperty("os.name");
+	static final String ENABLED_ON_CURRENT_OS = String.format("Enabled on operating system: %s (%s)",
```

Comment:
It would be much better! I just couldn't find a way to pass enabled/disabled reason to BooleanExecutionCondition as values that are calculated based on isEnabled call. 

I think I can refactor BooleanExecutionCondition to accept enabled/disabled reason as functions, but then isEnabled calculation would have to be called twice: actual isEnabled call & inside reason calculation. 

The other option would be to refactor isEnabled method signature so it returns a composite object with true/false & reason inside. 
```java
abstract EnablingResult isEnabled(A annotation);
```

However, it requires much more refactoring and will affect all BooleanExecutionCondition children. What do you think?

Reply:
`EnablingResult` sounds a lot like `ConditionEvaluationResult`, kinda like reinventing the wheel.

> However, it requires much more refactoring and will affect all BooleanExecutionCondition children. What do you think?

I don't think I'd make that change in a way that affects all `BooleanExecutionCondition` subclasses.

Instead, if you go that route, I'd probably introduce two variants of the abstract base class. For example, the unaffected subclasses could continue to extend `BooleanExecutionCondition`, and the OS variants could extend a new `AbstractOsBasedExecutionCondition` (or similar). And I probably wouldn't try too hard to avoid code duplication in the new abstract base class. If you can avoid some code duplication then good, but if you can't easily avoid code duplication I wouldn't worry about it too much. 

Disclaimer: this is just my personal opinion. 😉

Reply:
Hmm, I didn't realize we couldn't easily read the content of the annotation from here.

Now that I know, I don't think it's worth it to add all that complexity just to change the message. I would be ok keeping the original one, if y'all agree?

If we want to highlight the fact that OS and architecture are not necessarily enabled/disabled jointly, the message could be more like `Enabled on operating system %s and architecture %s`? But the usage of "and" might be even more confusing... 🤔

Reply:
We briefly discussed this in the team call and concluded that we think the clearer message justifies the added complexity.